File: Java Script - Introduction to Programming/8.2. Exam Preparation - Part II/04. Grades.js
function grades([...arg]) {
  arg = [...arg.map(Number)]
  let studentsCount = arg.shift()
  let sumGrades = arg.reduce((a, b) => a + b)

  console.log(`Top students: ${(arg.filter(el => el >= 5.00).length / studentsCount * 100).toFixed(2)}%`)
  console.log(`Between 4.00 and 4.99: ${(arg.filter(el => el >= 4.00 && el <= 4.99).length / studentsCount * 100).toFixed(2)}%`)
  console.log(`Between 3.00 and 3.99: ${(arg.filter(el => el >= 3.00 && el <= 3.99).length / studentsCount * 100).toFixed(2)}%`)
  console.log(`Fail: ${(arg.filter(el => el >= 2.00 && el <= 2.99).length / studentsCount * 100 ).toFixed(2)}%`)
  console.log(`Average: ${(sumGrades / studentsCount).toFixed(2)}`)
}

//grades(['10', '3.00', '2.99', '5.68', '3.01', '4', '4', '6.00', '4.50', '2.44', '5'])
//grades(['6', '2', '3', '4', '5', '6', '2.2'])

File: Pyrhon - Introduction to Programming/8.2. Exam Preparation - Part II/04. Grades.py
students_count = int(input())
arg = [float(input()) for _ in range(students_count)]
sum_grades = sum(arg)

print(f'Top students: {len(list(filter(lambda el: el >= 5.00, arg))) / students_count * 100:.2f}%')
print(f'Between 4.00 and 4.99: {len(list(filter(lambda el: 4.00 <= el <= 4.99, arg))) / students_count * 100:.2f}%')
print(f'Between 3.00 and 3.99: {len(list(filter(lambda el: 3.00 <= el <= 3.99, arg))) / students_count * 100:.2f}%')
print(f'Fail: {len(list(filter(lambda el: 2.00 <= el <= 2.99, arg))) / students_count * 100:.2f}%')
print(f'Average: {(sum_grades / students_count):.2f}')



'''
6
2
3
4
5
6
2.2

Top students: 33.33%
Between 4.00 and 4.99: 16.67%
Between 3.00 and 3.99: 16.67%
Fail: 33.33%
Average: 3.70
'''
